lib/matplotlib/backend_bases.py

```diff
@@ -2552,8 +2564,10 @@ def __init__(self, canvas, num):
         canvas.manager = self  # store a pointer to parent
         self.num = num
 
-        self.key_press_handler_id = self.canvas.mpl_connect('key_press_event',
-                                                            self.key_press)
+        if rcParams['toolbar'] != 'navigation':
+            self.key_press_handler_id = self.canvas.mpl_connect(
+                                                'key_press_event',
+                                                self.key_press)
```

Comment:
I am torn on this one, I see @fariza point that this is on it's way out, but I also don't like variable attributes on objects (just because you _can_ doesn't mean you _should_).  On the other hand, this is the _only_ place in our code base that we reference `self.key_press_handler_id` so  it probably does not really matter either way.

This would only be used to deregister a call back, does that code cope well with `None`?